Sources/SwiftParser/ParseSourceFile.swift

```diff
@@ -152,8 +152,25 @@ extension Parser {
     }
 
     let remainingTokens = self.consumeRemainingTokens()
+
+    func parseEndOfFileIfPresent(raw: RawSyntax) -> R? {
```

Comment:
I would make this a `private` function in the extension. That just seems cleaner to me. Nested functions always give me the feeling that they might be capturing outside state and I don’t look for them.

Also, I would name it something that explicitly mentions the leading trivia and add a doc comment.